File: bridge_sdks/go/msgrbridge/credentials.go
package msgrbridge

import (
	"context"
	"encoding/json"
	"os"
	"strings"
)

// CredentialBootstrapper prepares a daemon with credential material.
type CredentialBootstrapper interface {
	Bootstrap(ctx context.Context, service string) (map[string]any, error)
}

// EnvCredentialBootstrapper loads credentials from environment variables.
type EnvCredentialBootstrapper struct {
	Loader func(string) string
}

// NewEnvCredentialBootstrapper returns a bootstrapper using os.LookupEnv.
func NewEnvCredentialBootstrapper() EnvCredentialBootstrapper {
	return EnvCredentialBootstrapper{Loader: func(key string) string {
		return lookupEnv(key)
	}}
}

// Bootstrap implements CredentialBootstrapper.
func (b EnvCredentialBootstrapper) Bootstrap(ctx context.Context, service string) (map[string]any, error) {
	key := "MSGR_" + strings.ToUpper(service) + "_CREDENTIALS"
	raw := b.Loader(key)
	if raw == "" {
		return map[string]any{}, nil
	}

	var payload map[string]any
	if err := json.Unmarshal([]byte(raw), &payload); err != nil {
		return nil, err
	}

	return payload, nil
}

// lookupEnv is overridden in tests.
var lookupEnv = func(key string) string {
	value, _ := os.LookupEnv(key)
	return value
}

File: bridge_sdks/python/msgr_bridge_sdk/credentials.py
"""Credential bootstrap helpers for StoneMQ daemons."""

from __future__ import annotations

import json
import os
from typing import Mapping, Protocol


class CredentialBootstrapper(Protocol):
    async def bootstrap(self, service: str) -> Mapping[str, object]:
        """Return credential material for the given service."""


class EnvCredentialBootstrapper:
    """Loads credential JSON from environment variables."""

    def __init__(self, loader=None) -> None:
        self._loader = loader or os.getenv

    async def bootstrap(self, service: str) -> Mapping[str, object]:
        key = f"MSGR_{service.upper()}_CREDENTIALS"
        raw = self._loader(key)
        if not raw:
            return {}
        return json.loads(raw)
